hack/prom-rule-ci/prom-rules-tests.yaml

```diff
@@ -249,111 +249,126 @@ tests:
             exp_labels:
-              pod: "virt-controller-1"
               severity: "critical"
               kubernetes_operator_part_of: "kubevirt"
               kubernetes_operator_component: "kubevirt"
-      - eval_time: 5m
-        alertname: VirtOperatorRESTErrorsHigh
+      - eval_time: 20m
+        alertname: VirtHandlerRESTErrorsBurst
         exp_alerts:
           - exp_annotations:
-              summary: "More than 5% of the rest calls failed in virt-operator for the last hour"
-              runbook_url: "https://kubevirt.io/monitoring/runbooks/VirtOperatorRESTErrorsHigh"
+              summary: "More than 80% of the rest calls failed in virt-handler for the last 5 minutes"
+              runbook_url: "https://kubevirt.io/monitoring/runbooks/VirtHandlerRESTErrorsBurst"
             exp_labels:
-              pod: "virt-operator-1"
-              severity: "warning"
+              severity: "critical"
               kubernetes_operator_part_of: "kubevirt"
               kubernetes_operator_component: "kubevirt"
-      - eval_time: 5m
-        alertname: VirtOperatorRESTErrorsBurst
+      - eval_time: 20m
+        alertname: VirtApiRESTErrorsBurst
         exp_alerts:
           - exp_annotations:
-              summary: "More than 80% of the rest calls failed in virt-operator for the last 5 minutes"
-              runbook_url: "https://kubevirt.io/monitoring/runbooks/VirtOperatorRESTErrorsBurst"
+              summary: "More than 80% of the rest calls failed in virt-api for the last 5 minutes"
             exp_labels:
-              pod: "virt-operator-1"
               severity: "critical"
               kubernetes_operator_part_of: "kubevirt"
               kubernetes_operator_component: "kubevirt"
-      - eval_time: 5m
-        alertname: VirtHandlerRESTErrorsHigh
+
+
+  # High REST errors
+  - interval: 1m
+    input_series:
+      - series: 'rest_client_requests_total{namespace="ci", pod="virt-controller-1", code="200"}'
+        values: '0+10x120'
+      - series: 'rest_client_requests_total{namespace="ci", pod="virt-controller-1", code="400"}'
+        values: '0+100x120'
+      - series: 'rest_client_requests_total{namespace="ci", pod="virt-operator-1", code="200"}'
+        values: '0+10x120'
+      - series: 'rest_client_requests_total{namespace="ci", pod="virt-operator-1", code="400"}'
+        values: '0+100x120'
+      - series: 'rest_client_requests_total{namespace="ci", pod="virt-handler-1", code="200"}'
+        values: '0+10x120'
+      - series: 'rest_client_requests_total{namespace="ci", pod="virt-handler-1", code="500"}'
+        values: '0+100x120'
+      - series: 'rest_client_requests_total{namespace="ci", pod="virt-api-1", code="200"}'
+        values: '0+10x120'
+      - series: 'rest_client_requests_total{namespace="ci", pod="virt-api-1", code="500"}'
+        values: '0+100x120'
+
+    alert_rule_test:
+      - eval_time: 120m
```

Comment:
So to clarify, tell me if I got this right- the old test that's supposed to check for the percentage of bad requests in an hour actually simulated a run for 5 minutes? And it passed because our rules were wrong?

Reply:
exactly :)